.github/workflows/release.yaml

```
@@ -16,11 +16,57 @@ jobs:
       with:
         python-version: "3.11"
     - name: Install build backend
-      run: pip install hatch
+      run: pip install build
     - name: Build
-      run: hatch build -t wheel
-    - name: Publish
-      uses: softprops/action-gh-release@v1
+      run: python -m build
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v3
       with:
-        files: dist/digital_campaign_carbon_accounting-*.whl
-        draft: ${{ github.event_name == 'workflow_dispatch' }}
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-pypi:
+    name: >-
+      Publish Python distribution to PyPI
+    if: startsWith(github.ref, 'refs/tags/')  # only publish to PyPI on tag pushes
+    needs:
+    - build
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/digital-carbon-framework  # Replace <package-name> with your PyPI project name
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+    steps:
+    - name: Download all the dists
+      uses: actions/download-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Publish distribution 📦 to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+
+
+  publish-to-testpypi:
+    name: Publish Python 🐍 distribution 📦 to TestPyPI
+    needs:
+    - build
+    runs-on: ubuntu-latest
+
+    environment:
+      name: testpypi
+      url: https://test.pypi.org/p/digital-carbon-framework
+
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+
+    steps:
+    - name: Download all the dists
+      uses: actions/download-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Publish distribution 📦 to TestPyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        repository-url: https://test.pypi.org/legacy/
```

todo: you can merge these two jobs

```suggestion
env:
  pypi_url: ${{ startsWith(github.ref, 'refs/tags/') && 'https://pypi.org' || 'https://test.pypi.org/'

jobs
  [...]
  publish:
    name: ...
    needs: ...
    runs-on: ...
    environment:
      name: pypi
      url: "${{ env.pypi_url }}/p/digital-carbon-framework"
    permissions: ...
    steps:
    - name: Download all the dists
      uses: actions/download-artifact@v3
      with:
        name: python-package-distributions
        path: dist/
    - name: Publish distribution 📦
      uses: pypa/gh-action-pypi-publish@release/v1
      with:
        repository-url: ${{ env.pypi_url }}
```